broker-client/src/main/java/io/camunda/zeebe/broker/client/api/dto/BrokerRequest.java

```diff
@@ -74,22 +74,26 @@ public void serializeValue() {
 
   public abstract String getType();
 
+  public Optional<RequestDispatchStrategy> requestDispatchStrategy() {
```

Comment:
:exclamation: This here is "new" code. Essentially, before, the `BrokerRequestManager` was detecting that a request was a `BrokerPublishMessageRequest` (via an `instanceof`) and routing the request to the right partition based on the correlation key. 

Since the module doesn't know about this type anymore, I opted to change the logic of routing to:

- If a request has a key, then use the partition ID from that key.
- If it does not but requires a partition, then check if it provides its own request dispatch strategy. If it does, use it.
- If it doesn't provide anything, use the default request dispatch strategy (i.e. the round robin strategy).

You can see an example of this in `BrokerPublishMessageRequest` below.